makelib/common.mk

```diff
@@ -213,7 +213,9 @@ ifeq ($(shell git tag),)
 VERSION := $(shell echo "v0.0.0-$$(git rev-list HEAD --count)-g$$(git describe --dirty --always)" | sed 's/-/./2' | sed 's/-/./2' | sed 's/-/./2')
 else
 # use tags
-VERSION := $(shell git describe --dirty --always --tags | sed 's/-/./2' | sed 's/-/./2' | sed 's/-/./2' )
+# set if you want to consider only the tags matching your glob, e.g. "aws/*".
+VERSION_TAG_MATCH ?= '*'
```

Comment:
I think we also need to adop the `release.tag` make target.

Reply:
One thing I noticed is the following:
- Tag the HEAD with `azure/v0.2.0`
- Tag `HEAD^^` with `aws/v0.1.0`

```bash
> export VERSION_TAG_MATCH='azure/*'
> git describe --dirty --always --tags --match "${VERSION_TAG_MATCH}" | sed 's|.*/||' | sed 's/-/./2' | sed 's/-/./2' | sed 's/-/./2'

v0.2.0
```
as expected, but:

```bash
> export VERSION_TAG_MATCH='aws/*'
> git describe --dirty --always --tags --match "${VERSION_TAG_MATCH}" | sed 's|.*/||' | sed 's/-/./2' | sed 's/-/./2' | sed 's/-/./2'

v0.1.0-3.g398e02c3
```

Not sure what this implies as probably what's important for a release here is what we get from these commands in a _release branch_.

Reply:
> Not sure what this implies, as probably what's important for a release here is what we get from these commands in a release branch. Could you please check?

This is expected behavior. If there is any commit on top of the tagged one, it should return a suffixed version with the hash. We'd have a master build of AWS versioned as `v0.1.0-3.g398e02c3` in this case for example. In Crossplane, we add a new commit after the release-tagged one as `v0.2.0-rc` so that it's clear that's it's a pre-release version of `v0.2.0` of AWS, I think we can follow the same pattern. I'll add this flow to the design doc.

> I think we also need to adop the release.tag make target.

I just pushed a commit to add this but I think no one is using that target so we might as well just delete it since it adds a bit complexity to the code to handle that target as well. What do you think?

Reply:
Thanks for checking this.